python/private/python.bzl

```
@@ -104,7 +104,14 @@ def parse_modules(*, module_ctx, _fail = fail):
                 # * rules_python needs to set a soft default in case the root module doesn't,
                 #   e.g. if the root module doesn't use Python itself.
                 # * The root module is allowed to override the rules_python default.
-                is_default = toolchain_attr.is_default
+                if toolchain_attr.default_version_file == None:
+                    is_default = toolchain_attr.is_default
+                else:
+                    is_default = (
+                        module_ctx.read(toolchain_attr.default_version_file) == toolchain_version
+                    )
```

```suggestion
                if toolchain_attr.default_version_file:
                    version_from_file = module_ctx.read(toolchain_attr.default_version_file).strip()
                    is_default = version_from_file == toolchain_version
                else:
                    is_default = toolchain_attr.is_default
```

I think the inverted if reads a little bit easier and I also think we should have `.strip()` on the contents of the file to remove whitespace.